charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -36,6 +36,11 @@ spec:
         "vault.hashicorp.com/ca-cert": "/vault/custom/{{ .Values.global.secretsBackend.vault.ca.secretKey }}"
         {{- end }}
         {{- end }}
+        {{- if (and .Values.global.metrics.enabled .Values.global.metrics.enableGatewayMetrics) }}
```

Comment:
@Jeff-Apple the flags that we're keying off of can be found here: https://github.com/hashicorp/consul-k8s/blob/842452602717097c930e6fe6978f2c0416e31183/charts/consul/values.yaml#L441-L465

Question: Do we want to re-use `enableGatewayMetrics` for the purpose of this PR? If so, does it make sense to just add "API gateways" to the list of things that will expose metrics endpoints?

Reply:
I think that should be ok, since API GW is a superset of the Ingress GW uses the same metrics port.

Reply:
Committed minor docstring adjustment

Reply:
@david-yu So my question would be - do we have a pattern where our current `consul-k8s` controllers are exposing metrics via one of these flags? This particular change would only be exposing **_controller_** metrics, not individual gateway metrics, as we don't currently have some of the groundwork done for those yet (i.e. there's some work around proxying and metrics collection at the gateway deployment level that would need to be done to allow for that)

Reply:
@andrewstucki at the moment we don't expose controller metrics. This flag is for envoy metrics from those gateways. So in your case I think this flag would control the metrics emitted by the ingress gateways proxies that get created via CRDs.

I think if this is only for the controller metrics, then re-using this flag could be confusing. We don't have a flag for this at the moment, but maybe this could be something like  `controllerMetrics` which will then enable it for all controllers. We (the k8s team) will then need to also add this to our controllers.

Reply:
@ishustava great idea! I've added `enableControllerMetrics`. Can you please check the docstring for it and make sure that contract makes sense going foward?

Reply:
Looks good! thanks for updating so quickly. I've added one question about the port. I think we also need bats tests for this (sorry if this is something you're already aware of!)